src/mapping.ts

```diff
@@ -52,40 +54,70 @@ export function handleBlock(block: cosmos.Block): void {
 
 function saveBlock(id: string, b: cosmos.Block): void {
   const block = new Block(id);
-  block.header = saveHeader(id, b.header);
-  block.evidence = saveEvidenceList(id, b.evidence);
-  block.lastCommit = saveCommit(id, b.lastCommit);
-  block.resultBeginBlock = saveResponseBeginBlock(id, b.resultBeginBlock);
-  block.resultEndBlock = saveResponseEndBlock(id, b.resultEndBlock);
-  block.transactions = saveTxResults(id, b.transactions);
-  block.validatorUpdates = saveValidators(id, b.validatorUpdates);
+  block.header = saveHeader(`${id}-header`, b.header);
```

Comment:
I think it is pointless to save ids this way since the type of the structure is implicit in the query response. For example:

```
data": {
    "block": {
      "id": "5200793",
      "header": {
        "version": {
          "block": "11",
          "app": "0"
        },
        "chainId": "cosmoshub-4",
        "height": "5200793",
        "time": {
          "seconds": "1613651219",
          "nanos": 406056552
        },
.....
```

And when you are querying for a specific type you do it like this: header(id: "12345"). So I would remove them.

Reply:
We can't remove ids from graphql entity